File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */ 
class Solution {
public:
    void helper(TreeNode* root, vector<int>&nodesValues){
        if(!root){
            return;
        }
        nodesValues.push_back(root->val);
        helper(root->left, nodesValues);
        helper(root->right, nodesValues);
    } 
    vector<int> preorderTraversal(TreeNode* root) { 
        vector<int> nodesValues;
        helper(root, nodesValues);
        return nodesValues;
    };
};

File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ##node -> left -> right    
        self.nodes_values = []
        def dfs(root):
            if not root:
                return  
            self.nodes_values.append(root.val)
            dfs(root.left)
            dfs(root.right) 
        dfs(root) 
        return self.nodes_values
